### ISSUE TITLE
Is it possible to set an external PostgreSQL DB? For Clustering maybe?

### ISSUE THREAD

**Issue by mightymop** [Overv/openstreetmap-tile-server#21]
Hello, 

i'd like to know whether is it possible to set an external PostgreSQL database on an other host. 
This way multiple Tile Servers / Renderers could use the DB and save some storage.

**Comment by Istador** [comment]
In theory that should be possible.

I haven't done that, but you might only need to change a few config files or pass additional [environment variables](https://www.postgresql.org/docs/9.1/libpq-envars.html) to the container to make that happen.

But maybe you need to edit the [project.mml](https://github.com/gravitystorm/openstreetmap-carto/blob/master/project.mml) to set up your datasource and compile the `mapnik.xml` yourself.
``` bash
carto project.mml > mapnik.xml
```

See:
- [mapnik and postgresql in differents servers - OSM Help](https://help.openstreetmap.org/questions/51521/mapnik-and-postgresql-in-differents-servers)
- [How do I configure renderd and mod_tile for postgresql running on non default socket? - OSM Help](https://help.openstreetmap.org/questions/64223/how-do-i-configure-renderd-and-mod_tile-for-postgresql-running-on-non-default-socket)

---

If the database is on another host, that slows down rendering and therefore client response times.
It also creates a single point of failure and bottleneck, if all tile servers use the same database.

Note, that if you want to use the same database for different styles: they might require different things imported into the database.

**Comment by sadilet** [comment]
@mightymop Hi, did you connect to external postgres? If yes, could you give some instruction?

**Comment by icemagno** [comment]
Seems all guys working on OSM products ( like OSMEditor too ) have a huge desire in break some containerization premisses and put services and databases in same conteiner. Please... STOP DOING THIS.

**Comment by Istador** [comment]
> Seems all guys working on OSM products ( like OSMEditor too ) have a huge desire in break some containerization premisses and put services and databases in same conteiner. Please... STOP DOING THIS.

You're right, splitting it up would be better. But that'd probably be to complicated for 90% of the people using this image to setup, configure and maintain.

No one prevents you from doing this yourself.
